simplekv/cache.py

```diff
@@ -110,6 +110,26 @@ def open(self, key):
             # cache error, ignore completely and return from backend
             return self._dstore.open(key)
 
+    def _copy(self, source, dest):
+        """Implementation of :meth:`~simplekv.CopyMoveMixin.copy`.
+        
+        Copies the data in the backing store and removes the destination key from the cache,
+         in case it was already populated.
+         """
+        k = self._dstore._copy(source, dest)
+        self.cache.delete(dest)
```

Comment:
We could do this heuristically - if you equip a store with a copymovemixin and then pass it as backing store into a cachedecorator we use _copy/_move, else 2/3.
Or I abstain from my stance on using _copy/_move directly in the decorators and we rename the default implementations in CopyMoveMixin, so we can actually detect the presence of real backend implementations of copy and move.

After some considerations, I think this is the best way going forward.